### ISSUE TITLE
Offset error when using inline R code

### ISSUE THREAD

**Issue by ms609** [r-lib/roxygen2#1122]
I'm encountering an error when using inline R code to generate text for Rd files.
When the R string is in a list and as the start of a new line, the replacement of the R code block with its output is offset by two characters , leaving the "`r" visible and deleting
the subsequent two characters.

Minimal example:

`R/hello.R`
```r
#' Hello, world!
#'
#' @templateVar ok FALSE
#' @template template
hello <- function() {
  print("Hello, world!")
}
```

`man-roxygen/template.R`
```r
#' @details
#' - Roxygen exhibits
#' `r ifelse(<%= ok %>, "expected", "unexpected")`
#' behaviour
```

`DESCRIPTION` should contain `Roxygen: list(markdown = TRUE)`

Expected output:
```
- Roxygen exhibits unexpected behaviour
```

Actual output:
```
- Roxygen exhibits 'runexpectedehaviour
```

Workaround: Add spaces at start of lines in `template.R`:

`man-roxygen/template.R`
```r
#' @details
#' - Roxygen exhibits
#'   `r ifelse(<%= ok %>, "expected", "unexpected")`
#'   behaviour
```

**Comment by gaborcsardi** [comment]
I have never actually tried to use inline code in templates, so I am grad to hear that it works. :)

**Comment by hadley** [comment]
Also fixed by #1391
